include/zephyr/storage/flash_map.h

```diff
@@ -269,16 +271,72 @@ uint8_t flash_area_erased_val(const struct flash_area *fa);
 #define FLASH_AREA_SIZE(label) \
 	DT_REG_SIZE(DT_NODE_BY_FIXED_PARTITION_LABEL(label))
 
+/**
+ * Returns DTS node for label, if it is fixed-partition compatible
+ *
+ * @return DTS node with given label, if exists and is a child of
+ *	   a fixed-partition compatible.
+ */
+#define FIXED_PARTITION_NODE(label) \
+	COND_CODE_1(FIXED_PARTITION_EXISTS(DT_NODELABEL(label)), (DT_NODELABEL(label)), ())
```

Comment:
Don't you at least mean `COND_CODE_1(FIXED_PARTITION_EXISTS(label), (DT_NODELABEL(label)), ())`?

Below, `FIXED_PARTITION_EXISTS` already has a `DT_NODELABEL`, so this is expanding to `DT_FIXED_PARTITION_EXISTS(DT_NODELABEL(DT_NODELABEL(label))`, which can't be right.

Regarding the `()` in the "else" branch, do you maybe want to do `(DT_INVALID_NODE)` instead? Not sure what you need this for, but it might be useful to have an explicit signal that it's not a node.

Reply:
This should be removed at all. I have no use for it now and somehow I have left it in. Will remove the macro.